src/service/data-util.service.ts

```diff
@@ -122,17 +123,31 @@ export class JhiDataUtils {
         return size.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ' ') + ' bytes';
     }
 
-    setFileData(event, entity, field: string, isImage: boolean) {
-        if (event && event.target.files && event.target.files[0]) {
-            const file = event.target.files[0];
-            if (isImage && !/^image\//.test(file.type)) {
-                return;
+    /**
+     * Sets the base 64 data & file type of the 1st file on the event (event.target.files[0]) in the passed entity object
+     * and returns a promise.
+     *
+     * @param event the object containing the file (at event.target.files[0])
+     * @param entity the object to set the file's 'base 64 data' and 'file type' on
+     * @param field the field name to set the file's 'base 64 data' on
+     * @param isImage boolean representing if the file represented by the event is an image
+     * @returns a promise that resolves to the modified entity if operation is successful, otherwise rejects with an error message
+     */
+    setFileData(event, entity, field: string, isImage: boolean): Promise<any> {
+        return new Promise((resolve, reject) => {
+            if (event && event.target.files && event.target.files[0]) {
+                const file = event.target.files[0];
+                if (isImage && !/^image\//.test(file.type)) {
+                    reject(`File was expected to be an image but was found to be ${file.type}`);
+                } else {
+                    this.toBase64(file, (base64Data) => {
+                        entity[field] = base64Data;
+                        entity[`${field}ContentType`] = file.type;
+                        resolve(entity);
+                    });
+                }
             }
```

Comment:
please reject promise in else statement as I listed before

Reply:
and unit test for that as well

Reply:
done